tests/e2e/pageobjects/cell.page.ts

```diff
@@ -0,0 +1,229 @@
+import { BasePage } from 'wdio-vscode-service'
+import clipboard from 'clipboardy'
+
+const DEFAULT_SEARCH_FOR_CELL_TIMEOUT = 50000
+
+export enum OutputType {
+    TerminalView = 'terminal-view',
+    ShellOutput = 'shell-output',
+    Display = '.display'
+}
+
+export enum StatusBarElements {
+    Copy = 'Copy',
+    Configure = 'Configure',
+    CLI = 'CLI',
+    ShellScript = 'Shell Script',
+    OpenTerminal = 'Open Terminal',
+    BackgroundTask = 'Background Task',
+    StopTask = 'Stop Task'
+}
+
+export interface NotebookCommand {
+    element$: WebdriverIO.Element
+    text: string
+}
+
+export class Cell extends BasePage<{}, { cell: {} }> {
+    public locatorKey = 'cell' as const
+    #cellRow: WebdriverIO.Element
+    #cellText: string
+    constructor(cellRow: WebdriverIO.Element, cellText: string) {
+        // TODO: Write the locator map
+        super({ cell: {} })
+        this.#cellRow = cellRow
+        this.#cellText = cellText
+    }
+
+    /**
+     * Determines the output of a cell by identifying its successful execution status
+     * @param expectedContent The content used to identify the location of the code cell.
+     * @returns Promise<WebdriverIO.Element | undefined>
+     */
+    async #getCurrentExecutionRow(expectedContent?: string): Promise<WebdriverIO.Element | undefined> {
+        const successRows = await browser.$$('.codicon-notebook-state-success')
+        for await (const row of successRows) {
+            const executionRow = await row
+                .parentElement()
+                .parentElement()
+                .parentElement()
+                .parentElement()
+                .parentElement()
+            const cellEditor = await executionRow.$('.cell-editor-container')
+            const text = await cellEditor.getText()
+            if (text === (expectedContent ?? this.#cellText)) {
+                return executionRow
+            }
+        }
+    }
+
+    /**
+     * Finds the cell status bar command elements
+     * @returns Promise<NotebookCommand[]>
+     */
+    async #getStatusBarElements(): Promise<NotebookCommand[]> {
+        const executionRow = await this.#getCurrentExecutionRow()
+        const commands: NotebookCommand[] = []
+        if (executionRow) {
+            const statusBar$ = await executionRow.$('.cell-statusbar-container')
+            const statusRight$ = await statusBar$.$('.cell-status-right')
+            const contributedRight$ = await statusRight$.$('.cell-contributed-items-right')
+            const commandsResult$$ = await contributedRight$.$$('.cell-status-item-has-command')
+
+            for await (const row of commandsResult$$) {
+                const text = await row.getText()
+                commands.push({
+                    element$: row,
+                    text: text.trim()
+                })
+            }
+        }
+        return commands
```

Comment:
I prefer to return early and safe indentations, up to you:

```suggestion
        if (!executionRow) {
            return commands
        }

        const statusBar$ = await executionRow.$('.cell-statusbar-container')
        const statusRight$ = await statusBar$.$('.cell-status-right')
        const contributedRight$ = await statusRight$.$('.cell-contributed-items-right')
        const commandsResult$$ = await contributedRight$.$$('.cell-status-item-has-command')

        for await (const row of commandsResult$$) {
            const text = await row.getText()
            commands.push({
                element$: row,
                text: text.trim()
            })
        }

        return commands
```